### PR TITLE
feat(subagent): add max_time parameter for auto-cancel watchdog

### PR DESCRIPTION
**Pull request by TimeToBuildBob**
## Summary

- Adds `max_time: float | None = None` parameter to `subagent()` — fully backwards-compatible (defaults to no limit)
- Launches a daemon `threading.Timer` after all execution paths register the subagent; the watchdog fires `_timeout_subagent()` at deadline
- `_timeout_subagent()` uses `subagent_cancel()` (gptme/gptme#2889, already merged) with a race-safe check so a naturally-completed subagent is unaffected
- Adds `"timeout"` to the `Status` literal type; the completion hook emits ⏱️ instead of ❌
- 7 new unit tests covering: normal completion before deadline, timeout during execution, all three execution modes (subprocess/thread/acp)

## Motivation

`subagent_cancel()` was added in #2889 but callers must manually decide when to cancel — no built-in deadline. Without `max_time`, a stuck subagent blocks the parent agent indefinitely:

- Defensive orchestration: don't let one hung subagent stall a multi-agent pipeline
- CI-style hard budgets: "run this for at most 60s, return what you have"
- Autonomous sessions: Bob's 50-minute session limit can now be enforced per-subagent

## Test plan

- [ ] `pytest tests/test_subagent_unit.py -v -k MaxTime` passes (7 new tests)
- [ ] `max_time=None` (default) path: no regression in existing subagent tests
- [ ] Subprocess-mode: SIGTERM/SIGKILL on timeout
- [ ] Thread-mode: marked timed-out at next checkpoint
- [ ] ACP-mode: cancel call delivered to ACP client

### CODE REVIEW

**Comment by TimeToBuildBob** [comment]
@greptileai review

**Comment by TimeToBuildBob** [comment]
Fixed the two P1 issues Greptile flagged:

1. **`max_time` dropped on clarification re-spawn** — `subagent_reply()` now passes `max_time=sa.max_time` to the re-spawned `subagent()` call, so a subagent that requests clarification inherits its original deadline instead of running indefinitely.

2. **`max_time` silently ignored in planner mode** — `_run_planner()` now accepts a `max_time` parameter (forwarded from `subagent()`) and starts a `threading.Timer` watchdog for each executor subagent after registration, covering both subprocess and thread execution modes. Used a deferred import to avoid the `api.py → execution.py → api.py` circular import.

Also updated the `subagent_reply` parameter-preservation test to include `max_time` in the expected kwargs dict. All 121 unit tests pass.

**Comment by TimeToBuildBob** [comment]
Addressed both P1 issues flagged by Greptile:

1. **`max_time` dropped on clarification re-spawn** (`subagent_reply`): Added `max_time=sa.max_time` to the `subagent()` re-spawn call, so the time budget is preserved across clarification restarts.

2. **`max_time` silently ignored in planner mode**: Restructured the `mode="planner"` block to register the Subagent and launch the watchdog timer before calling `_run_planner`, and cancel it in a `finally` block on completion. Planner subagents now get the same timeout behavior as thread/subprocess/ACP modes.

Also added a test assertion for `max_time=None` in the clarification re-spawn test to match the new parameter.

✅ `pytest tests/test_subagent_unit.py`: 121/121 passing

File: gptme/tools/subagent/api.py [hunk 1]
```diff
@@ -677,11 +688,58 @@ def run_subagent():
...
+        logger.info(
+            f"Subagent '{agent_id}' subprocess killed after {max_time}s (max_time)."
+        )
+    else:
+        logger.info(
+            f"Subagent '{agent_id}' marked timed-out after {max_time}s "
+            "(thread will stop at its next checkpoint)."
+        )
+
+    notify_completion(agent_id, "timeout", f"Timed out after {max_time}s")
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply gptme/tools/subagent/api.py:776]: Acknowledged — this is a documented P2 limitation. ACP threads block inside `asyncio.run(_acp_run())` until the remote server responds; there's no cancellation primitive at the ACP client layer today.

The daemon-thread flag ensures the ACP thread won't prevent process exit, and the result slot is already marked `timeout` so the parent loop receives the notification immediately. The ACP server eventually tears down the session on its end.

A proper fix (cooperative cancellation via an asyncio Event or ACP cancel-request) is worth a follow-up, but is out of scope for this patch. I've noted it as a known limitation in the docstring.

**Comment by TimeToBuildBob** [comment]
@greptileai review

**Comment by ErikBjare** [comment]
@TimeToBuildBob Merged with a Greptile 3/5, seems to have been some race or mistake by a LLM that prematurely merged. Fix it and attribute the harm.
